src/util/small_map.h

```diff
@@ -190,35 +190,35 @@ class small_mapt
   static_assert(S_BITS <= N_BITS, "S_BITS should be no larger than N_BITS");
 
   static_assert(
-    std::numeric_limits<unsigned>::digits >= BITS,
+    std::numeric_limits<std::size_t>::digits >= BITS,
     "BITS must fit into an unsigned");
 
   // Internal
 
-  unsigned get_field(std::size_t field) const
+  std::size_t get_field(std::size_t field) const
   {
     PRECONDITION(field < NUM);
 
-    unsigned shift = field * BITS;
+    std::size_t shift = field * BITS;
     return (ind & (MASK << shift)) >> shift;
   }
 
-  void set_field(std::size_t field, unsigned v)
+  void set_field(std::size_t field, std::size_t v)
   {
     PRECONDITION(field < NUM);
     PRECONDITION((std::size_t)(v >> 1) < NUM);
 
-    unsigned shift = field * BITS;
+    std::size_t shift = field * BITS;
 
     ind &= ~((index_fieldt)MASK << shift);
-    ind |= v << shift;
+    ind |= (index_fieldt)v << shift;
```

Comment:
It is guaranteed by the preconditions of the method and the static assertions that `index_fieldt` is large enough to hold `v << shift`.